### DISCUSSION TITLE
Support Airtable views

### DISCUSSION THREAD

**Discussion by pjmv** [supabase/supabase#26641] [Contribute to Supabase]
Airtable allows the users to define views that can incorporate filters.

The Airtable wrapper should support using a view when defining a foreign table. This would allow some filtering and data integrity control (something Airtable is generally very bad for) before it is mirrored in postgres.

**Comment by TheOtherBrian1** [comment]
I've come to realize that views in PostgreSQL are different from views in Airtable.

I found the following in the Airtable [FDW code](https://github.com/supabase/wrappers/blob/182b165aeb1739e094945ae22f6486224c19b404/wrappers/src/fdw/airtable_fdw/airtable_fdw.rs#L123):

> `// TODO: Support specifying other options (view)`

It seems that the maintainers considered this feature, but have left it on the back burner. To get more attention, you may want to raise it as an issue with the label "enhancement".

**Reply by pjmv** [reply]
Thanks for the hint. I had stumbled on that comment in the sourcecode as well which prompted me to come here, but I wasn't sure which channel is the most effective to get something across... Between Discord, the supabase/supabase discussions, supabase/supabase issues, supabase/wrappers discussions and supabase/wrappers issues... Unsure what is going to be the most effective :)

The main reason I see value in having this feature implemented is because Airtable does not provide effective validation mechanisms for row creation, so in practice we often have bad rows in our Airtable tables. Airtable views are effective to weed out those rows and such an option would allow the user to have clean rows straight from the foreign table in postgres.

Obviously doing the filtering in a sql view downstream is easy as well so this is kinda low criticity

**Comment by burmecia** [comment]
Thanks for the request! Airtable View support is nice feature and has been added by this PR #134.

**Reply by pjmv** [reply]
Nice ! Thanks a lot !
